AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/StorageEngine.swift

```diff
@@ -261,6 +255,35 @@ final class StorageEngine: StorageEngineBehavior {
         }
     }
 
+    private func delete<M: Model>(_ modelType: M.Type, predicate: QueryPredicate) -> DataStoreResult<[M]> {
```

Comment:
if we change this private function named to `queryAndDeleteTransaction`, then it's calling code would be more readable:
```
        let transactionResult = queryAndDeleteTransaction(modelType, predicate: predicate)
```
(where currently we have:
```
        let transactionResult = delete(modelType, predicate: predicate)
```